### PR TITLE
Migrate react naviagion library

### PR DESCRIPTION
**Pull request by Kamill90**

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -16,7 +16,8 @@ export const HEADER_HEIGHT = Platform.select({
   android: 38,
 }) as number;
 
-interface Props extends Partial<NavigationScreenProps> {
+interface Props {
+  navigation?: StackNavigationProp<any, any> | any;
```
Comment (bklukaczewski) [inline src/components/Header.tsx:20]: about TypeScript, I know it's still WIP but consider using alias type for `navigation` somewhere in your code
like:
```
type NavigationProp = StackNavigationProp<any, any> | any;
```
and use it in your components. 
Having this additional abstraction layer should help with refactoring in the future.

File: src/components/ScreenTemplate.tsx [hunk 2]
```diff
@@ -61,22 +65,20 @@ export class ScreenTemplate extends React.PureComponent<Props> {
...
+      </SafeAreaProvider>
     );
   }
 }
 
 const styles = StyleSheet.create({
   container: {
     flex: 1,
+    backgroundColor: palette.white,
+    paddingBottom: ifIphoneX(20, 0),
```
Comment (bklukaczewski) [inline src/components/ScreenTemplate.tsx:81]: just to make sure: Bottom notch height for iPhoneX is 34 by design :)

File: src/consts/models.tsx [hunk 3]
```diff
@@ -116,3 +127,109 @@ export interface Filters {
...
+    value?: string;
+    validate?: (value: string) => string | undefined;
+    keyboardType?: KeyboardType;
+  };
+};
+
+export type MainCardStackNavigatorParamList = {
+  [Route.MainCardStackNavigator]: undefined;
+  [Route.CreateWallet]: undefined;
+  [Route.ImportWallet]: { appSettings: AppSettingsState; loadWallets: () => Promise<WalletsActionType> };
```
Comment (bklukaczewski) [inline src/consts/models.tsx:198]: I'm not really a big fan of such formatting with prettier. I believe we should talk about always making larger objects like
```
{
  appSettings: AppSettingsState; 
  loadWallets: () => Promise<WalletsActionType>;
};

There's a rule for that I guess, just need to decide if we want to apply it or not :)

File: src/consts/models.tsx [hunk 4]
```diff
@@ -116,3 +127,109 @@ export interface Filters {
...
   transactionType?: string;
 }
+
+export type MainTabNavigatorParamList = {
+  [Route.Dashboard]: undefined;
+  [Route.ContactList]: undefined;
+  [Route.Settings]: undefined;
+};
+
+export type RootStackParamList = {
```
Comment (bklukaczewski) [inline src/consts/models.tsx:137]: technically it's not a list 😀

File: src/navigators/EditTextNavigator.tsx [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-  {
-    EditText: EditTextScreen,
-  },
-  {
-    headerMode: 'screen',
-    mode: 'modal',
-  },
+const Stack = createStackNavigator();
+
+export const EditTextNavigator = () => (
```
Comment (bklukaczewski) [inline src/navigators/EditTextNavigator.tsx:9]: I would consider making the structure of views / navigators much more flat (remove unnecessary navigators) and customize their animations per screen
```
const modalTransition = {
  cardStyleInterpolator: CardStyleInterpolators.forVerticalIOS,
  gestureDirection: 'vertical' as any,
};

...
<Stack.Screen
          name={Route.CreateAdSuccess}
          component={CreateAdSuccessScreen}
          options={{ ...modalTransition }}
        />
```

File: src/navigators/MessageNavigator.tsx [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
-  {
-    [Route.Message]: MessageScreen,
-  },
-  {
-    headerMode: 'none',
-    mode: 'modal',
-  },
+const Stack = createStackNavigator();
+
+export const MessageNavigator = () => (
```
Comment (bklukaczewski) [inline src/navigators/MessageNavigator.tsx:9]: ditto about removing unnecessary navigators / customising animations per screen

File: src/consts/models.tsx [hunk 7]
```diff
@@ -116,3 +127,109 @@ export interface Filters {
...
+    asyncTask?: () => void;
+  };
+  [Route.ExportWallet]: { wallet: Wallet };
+  [Route.ExportWalletXpub]: { wallet: Wallet };
+  [Route.DeleteWallet]: { wallet: Wallet };
+  [Route.DeleteContact]: { contact: Contact };
+  [Route.MainCardStackNavigator]: undefined;
+  [Route.SendTransactionDetails]: {
+    fee: number;
+    recipients: any;
```
[the remark this answers is not in the document]
Comment (Kamill90) [reply src/consts/models.tsx:167]: does tx always equal Transaction type?

File: src/screens/PinFlow/CreatePinScreen.tsx [hunk 8]
```diff
@@ -45,16 +32,22 @@ export class CreatePinScreen extends PureComponent<Props, State> {
 
   pinInputRef = React.createRef<PinInput>();
   backHandler?: NativeEventSubscription;
+  focusListener: any;
```
[the remark this answers is not in the document]
Comment (Kamill90) [reply src/screens/PinFlow/CreatePinScreen.tsx:35]: interesting, thx

File: src/screens/PinFlow/CreateTransactionPassword.tsx [hunk 9]
```diff
@@ -51,7 +52,7 @@ export class CreateTransactionPassword extends PureComponent<Props, State> {
 
   onSave = () => {
     this.props.navigation.navigate(Route.ConfirmTransactionPassword, {
-      password: this.state.password,
+      setPassword: this.state.password,
```
[the remark this answers is not in the document]
Comment (Kamill90) [reply src/screens/PinFlow/CreateTransactionPassword.tsx:55]: due to no-shadow eslint rule. `alreadySetPassword` would be better. Any naming suggestions are more than welcome.

File: src/consts/models.tsx (hunk 4 above)
Comment (Kamill90) [reply src/consts/models.tsx:137]: Yes but I wanted to keep it consistent with official documentation

**Pull request merged**